[vault operator] mlock and third party plugin

**Is your feature request related to a problem? Please describe.**
I'm not able to run properly a vault container with a third party plugin with mlock enabled. Whereas the vault binary is set to work with mlock by default (see vault Dockerfile line 90 https://github.com/hashicorp/docker-vault/blob/master/0.X/docker-entrypoint.sh), it's not the case for plugins.

That's why when I try to mount a third party plugin I got the following log in the vault container:
`2020-12-02T14:01:39.395Z [DEBUG] secrets.orchestrate.orchestrate_f699d532.orchestrate.orchestrate: 2020/12/02 14:01:39 cannot allocate memory: metadata=true
2020-12-02T14:01:39.396Z [ERROR] secrets.system.system_3eb3e496: error occurred during enable mount: path=orchestrate/ error="Unrecognized remote plugin message: 
This usually means that the plugin is either invalid or simply
needs to be recompiled to support the latest protocol."`

I got **cannot allocate memory** which clearly shows that my plugin is not able to use the `mlock` syscall as mentioned https://www.vaultproject.io/docs/internals/plugins.

So far I found 2 imperfect solutions to make my plugin work:
1. Add disable_mlock: true in the vault config but it's not recommended in production.
2. Manually run setcap cap_ipc_lock=+ep /vault/plugins/orchestrate inside my container (via kubectl exec) but its a manual procedure.

**Describe the solution you'd like**
Able to pass commands in the vault CRD to be executed when the vault container is created. So that I could `setcap` my third party plugin.
This command could be injected as PostStart in the vault lifecycle hook. https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/


**Additional context**
Here is the Vault CR I applied
`  apiVersion: "vault.banzaicloud.com/v1alpha1"
  kind: "Vault"
  metadata:
    name: "vault"
  spec:
    size: 1
    image: vault:1.6.0
    bankVaultsImage: banzaicloud/bank-vaults:1.8.0
...
    externalConfig
      secrets:
        - path: orchestrate
          type: plugin
          plugin_name: orchestrate
          description: Orchestrate key storage gateway
          config:
            force_no_cache: true
            passthrough_request_headers: ["X-Vault-Namespace"]
      plugins:
        - plugin_name: orchestrate
          command: orchestrate
          sha256: 5d63d9891463c8b7dc281759c105b45835bc8e91cde019a9bde74d858f795740
          type: secret
...`

--- a/operator/deploy/cr.yaml
+++ b/operator/deploy/cr.yaml
@@ -206,6 +206,14 @@ spec:
   #     hostnames:
   #     - "vault.local"
 
+  # It is possible to override the Vault container directly:
+  # vaultContainerSpec:
+  #   lifecycle:
+  #     postStart:
+  #       exec:
+  #         command:
+  #              - setcap cap_ipc_lock=+ep /vault/plugins/orchestrate
+  
   # Marks presence of Istio, which influences things like port namings
   istioEnabled: false
 

--- a/operator/pkg/apis/vault/v1alpha1/vault_types.go
+++ b/operator/pkg/apis/vault/v1alpha1/vault_types.go
@@ -188,6 +188,11 @@ type VaultSpec struct {
 	// default:
 	VaultPodSpec v1.PodSpec `json:"vaultPodSpec"`
 
+	// VaultContainerSpec is a Kubernetes Container specification snippet that will be merged into the operator generated
+	// Vault Container specification.
+	// default:
+	VaultContainerSpec v1.Container `json:"vaultContainerSpec"`
+
 	// VaultConfigurerAnnotations define a set of Kubernetes annotations that will be added to the Vault Configurer Pod.
 	// default:
 	VaultConfigurerAnnotations map[string]string `json:"vaultConfigurerAnnotations"`

--- a/operator/pkg/apis/vault/v1alpha1/zz_generated.deepcopy.go
+++ b/operator/pkg/apis/vault/v1alpha1/zz_generated.deepcopy.go
@@ -411,6 +411,7 @@ func (in *VaultSpec) DeepCopyInto(out *VaultSpec) {
 		}
 	}
 	in.VaultPodSpec.DeepCopyInto(&out.VaultPodSpec)
+	in.VaultContainerSpec.DeepCopyInto(&out.VaultContainerSpec)
 	if in.VaultConfigurerAnnotations != nil {
 		in, out := &in.VaultConfigurerAnnotations, &out.VaultConfigurerAnnotations
 		*out = make(map[string]string, len(*in))

--- a/operator/pkg/controller/vault/vault_controller.go
+++ b/operator/pkg/controller/vault/vault_controller.go
@@ -1403,6 +1403,12 @@ func statefulSetForVault(v *vaultv1alpha1.Vault, externalSecretsToWatchItems []c
 		return nil, err
 	}
 
+	// merge provided VaultContainerSpec into the Vault Container defined above
+	// the values in VaultContainerSpec will never overwrite fields defined in the PodSpec above
+	if err := mergo.Merge(&podSpec.Containers[0], v.Spec.VaultContainerSpec, mergo.WithOverride); err != nil {
+		return nil, err
+	}
+
 	podManagementPolicy := appsv1.ParallelPodManagement
 	if v.Spec.IsRaftStorage() || v.Spec.IsRaftHAStorage() {
 		podManagementPolicy = appsv1.OrderedReadyPodManagement

